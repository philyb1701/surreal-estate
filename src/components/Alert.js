import React from "react";
import PropTypes from "prop-types";
import "../styles/alert.css";

const Alert = ({ message, success }) => {
  if (!message) return null;

  return (
    <div className={`alert alert-${success ? "success" : "error"}`}>
      <p>{message}</p>
    </div>
  );
};

Alert.defaultProps = {
  success: false,
};

Alert.propTypes = {
  message: PropTypes.string.isRequired,
  success: PropTypes.bool,
};

export default Alert;
