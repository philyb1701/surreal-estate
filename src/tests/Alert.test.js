import React from "react";
import { render } from "@testing-library/react";
import Alert from "../components/Alert";

it("displays an Error message", () => {
  const { getByText, asFragment } = render(<Alert message="Error!" />);

  expect(getByText(/Error/).textContent).toBe("Error!");
  expect(asFragment()).toMatchSnapshot();
});

it("displays a Success message", () => {
  const { getByText, asFragment } = render(
    <Alert message="Success!!!" success />
  );

  expect(getByText(/Success/).textContent).toBe("Success!!!");
  expect(asFragment()).toMatchSnapshot();
});

it("displays nothing, if message prop is empty", () => {
  const { asFragment } = render(<Alert message="" />);
  const alert = asFragment();

  expect(alert).toMatchSnapshot();
});
