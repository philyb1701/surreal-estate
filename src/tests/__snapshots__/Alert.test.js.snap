// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays a Success message 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
  >
    <p>
      Success!!!
    </p>
  </div>
</DocumentFragment>
`;

exports[`displays an Error message 1`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  >
    <p>
      Error!
    </p>
  </div>
</DocumentFragment>
`;

exports[`displays nothing, if message prop is empty 1`] = `<DocumentFragment />`;
